src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -31,28 +35,77 @@ public class Engine : IPowerFxEngine
         /// <param name="powerFxConfig"></param>
         public Engine(PowerFxConfig powerFxConfig)
         {
-            powerFxConfig.Lock();
             Config = powerFxConfig;
         }
 
+        // All functions that powerfx core knows about. 
+        // Derived engines may only support a subset of these builtins, 
+        // and they may add their own custom ones. 
+        private static readonly ReadOnlySymbolTable _allBuiltinCoreFunctions = ReadOnlySymbolTable.NewDefault(Core.Texl.BuiltinFunctionsCore.BuiltinFunctionsLibrary);
+
+        /// <summary>
+        /// Builtin functions supported by this engine. 
+        /// </summary>
+        public ReadOnlySymbolTable SupportedFunctions { get; protected internal set; } = _allBuiltinCoreFunctions;
+
+        // By default, we pull the core functions. 
+        // These can be overridden. 
+        internal IEnumerable<TexlFunction> Functions => CreateResolverInternal().Functions;
+
+        /// <summary>
+        /// Get all functions from the config and symbol tables. 
+        /// </summary>
+        public IEnumerable<FunctionInfo> FunctionInfos => Functions.Select(f => new FunctionInfo(f));
+
         /// <summary>
         /// List all functions (both builtin and custom) registered with this evaluator. 
         /// </summary>
         public IEnumerable<string> GetAllFunctionNames()
         {
-            return Config.GetAllFunctionNames();
+            return FunctionInfos.Select(func => func.Name).Distinct();
+        }
+
+        // Additional symbols for the engine.
+        // A derived engine can replace this completely to inject engine-specific virtuals. 
+        // These symbols then feed into the resolver
+        protected SymbolTable EngineSymbols { get; set; }
+
+        // Move away
+        
```

Comment:
nit: is this a todo comment, or something else?

Reply:
I think this is a merge  conflict - I should remove altogether.